src/ripple/app/tests/Env.cpp

```diff
@@ -0,0 +1,217 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2012-2015 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/app/tests/Env.h>
+#include <ripple/app/paths/FindPaths.h>
+#include <ripple/app/tx/TransactionEngine.h>
+#include <ripple/basics/Slice.h>
+#include <ripple/json/to_string.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <ripple/protocol/HashPrefix.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/LedgerFormats.h>
+#include <ripple/protocol/Serializer.h>
+#include <ripple/protocol/STParsedJSON.h>
+#include <ripple/protocol/SystemParameters.h>
+#include <ripple/protocol/TER.h>
+#include <ripple/protocol/TxFlags.h>
+// VFALCO TODO Use AnyPublicKey, AnySecretKey, AccountID
+
+namespace ripple {
+namespace test {
+
+STAmount
+AccountInfo::balance(
+    Issue const& issue) const
+{
+    if (isXRP(issue))
+        return root_->getFieldAmount(sfBalance);
```

Comment:
If you try to get the `balance` (or really any other info) of an unfunded account, `root_` will be null, which will cause an exception to be thrown. Is this intentional, or should these functions be checking `root_` first?